Description: Additional changes to work with the Debian package of singular
 This patch needs some more work, at least the hardcoded multiarch path
 of libsingular...

--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -467,7 +467,7 @@
 
 if [ "$1" = '-singular' -o "$1" = '--singular' ]; then
     shift
-    exec singular "$@"
+    exec Singular "$@"
 fi
 
 if [ "$1" = '-sqlite3' -o "$1" = '--sqlite3' ]; then
--- a/sage/src/sage/interfaces/singular.py
+++ b/sage/src/sage/interfaces/singular.py
@@ -2272,7 +2272,7 @@
     nodes.clear()
     node_names.clear()
 
-    singular_docdir = os.environ["SAGE_LOCAL"]+"/share/singular/"
+    singular_docdir = "/usr/share/singular/"
 
     new_node = re.compile("File: singular\.hlp,  Node: ([^,]*),.*")
     new_lookup = re.compile("\* ([^:]*):*([^.]*)\..*")
--- a/sage/src/sage/libs/singular/singular.pyx
+++ b/sage/src/sage/libs/singular/singular.pyx
@@ -771,7 +771,7 @@
         extension = "so"
 
     # library name changed from libsingular to libSingular btw 3.x and 4.x
-    lib = os.environ['SAGE_LOCAL']+"/lib/libSingular."+extension
+    lib = "/usr/lib/x86_64-linux-gnu/libsingular-Singular.so"
 
     if not os.path.exists(lib):
         raise ImportError("cannot locate Singular library ({})".format(lib))
--- a/sage/src/sage/misc/cython.py
+++ b/sage/src/sage/misc/cython.py
@@ -306,8 +306,8 @@
 
         sage: code = [
         ... "#clang C++",
-        ... "#cinclude %s/include/singular %s/include/factory"%(SAGE_LOCAL, SAGE_LOCAL),
-        ... "#clib m readline Singular givaro ntl gmpxx gmp",
+        ... "#cinclude /usr/include/singular /usr/include/singular/singular /usr/include/x86_64-linux-gnu/singular /usr/include/x86_64-linux-gnu/singular/singular",
+        ... "#clib m readline singular-Singular givaro ntl gmpxx gmp",
         ... "from sage.rings.polynomial.multi_polynomial_libsingular cimport MPolynomial_libsingular",
         ... "from sage.libs.singular.polynomial cimport singular_polynomial_pow",
         ... "def test(MPolynomial_libsingular p):",
--- a/sage/src/sage/interfaces/tests.py
+++ b/sage/src/sage/interfaces/tests.py
@@ -39,7 +39,7 @@
     0
     sage: subprocess.call("echo syntax error | ipython", **kwds) in (0,1)
     True
-    sage: subprocess.call("echo syntax error | singular", **kwds)
+    sage: subprocess.call("echo syntax error | Singular", **kwds)
     0
 """
 from __future__ import print_function
